### ISSUE TITLE
boosted tree classifier h-params

### ISSUE THREAD

**Issue by samanemami** [tensorflow/tensorflow#48717]
<em>I am working with the [boosted tree classifier](https://github.com/tensorflow/estimator/blob/781c0d30c6bf100aa174591dd97cb70fc39d294d/tensorflow_estimator/python/estimator/canned/boosted_trees.py#L1933)</em>

**System information**
- TensorFlow installed from: pip
- TensorFlow version: 2.4.1
- Python version: 3.6.5

**Describe the current behavior**
There is a hyper-parameter in the model called `n_trees` which its 
default value is 100, also, there is a step parameter `max_step=100` in the training method.

**Describe the expected behavior**
Based on the gradient boosting classifier, `n_trees` refers to the number of trees, which we add at each boosting iteration.
The question is, what is the difference between these two parameters in the model?

**Standalone code to reproduce the issue**
I tried to plot the boosting accuracy and have the accuracy by adding a new tree at each iteration [here](https://github.com/samanemami/TFBoostedTree/blob/main/examples/Accuracy_plot.ipynb). 
I used `max_step` to have boosting iteration, so what is the **role of `max_step`** (in train method)?

**Other info / logs** Include any logs or source code that would be helpful to
[Boosted tree](https://github.com/tensorflow/estimator/blob/781c0d30c6bf100aa174591dd97cb70fc39d294d/tensorflow_estimator/python/estimator/canned/boosted_trees.py#L1933).
[train method](https://github.com/tensorflow/estimator/blob/781c0d30c6bf100aa174591dd97cb70fc39d294d/tensorflow_estimator/python/estimator/canned/boosted_trees_test.py#L403).
